source/m2mconnectionhandlerpimpl.cpp

```diff
@@ -216,15 +216,27 @@ void M2MConnectionHandlerPimpl::dns_handler()
             // Initialize the socket to stable state
             close_socket();
 
-            if(PAL_SUCCESS != pal_getAddressInfo(_server_address.c_str(), &_socket_address, &_socket_address_len)){
+            status = pal_getAddressInfo(_server_address.c_str(), &_socket_address, &_socket_address_len);
+            if (PAL_SUCCESS != status) {
+                tr_error("addrInfo, err: %d", (int)status);
+                // XXX: the error code given to client is wrong, a DNS_RESOLVING_ERROR might be better here.
                 _observer.socket_error(M2MConnectionHandler::SOCKET_ABORT);
```

Comment:
There is a DNS_RESOLVING_ERROR which should be used here, can you fix it in this PR. ?